<a href="https://colab.research.google.com/github/KARTIKPARATKAR/MACHINE-LEARNING-WORK/blob/main/TrainTestSplitTheData.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Machine Learning Project WorkFlow:

1)Getting the appropriate data.

2)Data Preprocessing

3)Data analysis like plotting the data

4)Train-Test-Split

5)Feeding data to a machine learning model

6)Training of model will be done by train data and evaluation of the model is

 done by test data.



In [33]:
import numpy as np
import pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split
from sklearn import svm                                      #Here svm is a support vector machine model
from sklearn.metrics import accuracy_score

In [34]:
#Link of the dataset used is - https://www.dropbox.com/s/uh7o7uyeghqkhoy/diabetes.csv?e=1&dl=0

In [35]:
diabetesdataset=pd.read_csv('/content/diabetes.csv')

In [36]:
diabetesdataset.head()  #Printing 5 rows of the diabetesdataset

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [37]:
diabetesdataset.shape  #This means that there are 768 rows and 9 columns present in the dataset

(768, 9)

In [38]:
#Getting the statistical measures of the data
diabetesdataset.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [39]:
diabetesdataset['Outcome'].value_counts()  #There are 500 labels of 0 and 268 labels of 1

0    500
1    268
Name: Outcome, dtype: int64

In [40]:
diabetesdataset.groupby('Outcome').mean()   #This is a mean of each column for label 0 and label 1.

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
Outcome                                                                      
0           3.298000  109.980000      68.184000      19.664000   68.792000   
1           4.865672  141.257463      70.824627      22.164179  100.335821   

               BMI  DiabetesPedigreeFunction        Age  
Outcome                                                  
0        30.304200                  0.429734  31.190000  
1        35.142537                  0.550500  37.067164

In [41]:
#Here we are splitting dataset into feature and target column. Features are those all columns except Outcome. And target column is Output
X=diabetesdataset.drop(columns='Outcome',axis=1)
Y=diabetesdataset['Outcome']

In [42]:
print(X)  #Printing X which are the feature columns

     Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0              6      148             72             35        0  33.6   
1              1       85             66             29        0  26.6   
2              8      183             64              0        0  23.3   
3              1       89             66             23       94  28.1   
4              0      137             40             35      168  43.1   
..           ...      ...            ...            ...      ...   ...   
763           10      101             76             48      180  32.9   
764            2      122             70             27        0  36.8   
765            5      121             72             23      112  26.2   
766            1      126             60              0        0  30.1   
767            1       93             70             31        0  30.4   

     DiabetesPedigreeFunction  Age  
0                       0.627   50  
1                       0.351   31  


In [43]:
print(Y)  # Printing Y which is a target column

0      1
1      0
2      1
3      0
4      1
      ..
763    0
764    0
765    0
766    1
767    0
Name: Outcome, Length: 768, dtype: int64


In [44]:
#Now performing data standardization
scaler= StandardScaler()  #Data standardization is used to make all column values in a perticular standard range.

In [45]:
scaler.fit(X)

StandardScaler()

In [46]:
standardizeddata=scaler.transform(X)

In [47]:
print(standardizeddata)

[[ 0.63994726  0.84832379  0.14964075 ...  0.20401277  0.46849198
   1.4259954 ]
 [-0.84488505 -1.12339636 -0.16054575 ... -0.68442195 -0.36506078
  -0.19067191]
 [ 1.23388019  1.94372388 -0.26394125 ... -1.10325546  0.60439732
  -0.10558415]
 ...
 [ 0.3429808   0.00330087  0.14964075 ... -0.73518964 -0.68519336
  -0.27575966]
 [-0.84488505  0.1597866  -0.47073225 ... -0.24020459 -0.37110101
   1.17073215]
 [-0.84488505 -0.8730192   0.04624525 ... -0.20212881 -0.47378505
  -0.87137393]]


In [48]:
X=standardizeddata
Y=diabetesdataset['Outcome']

Now splitting the data into training data and test data using train_test_split function.

In [49]:
X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size=0.2,random_state=2)  #Here test_size =0.2 means that 20% data will be used for testing the model and remaining 80 of data will be used for training the model.

In [50]:
print(X.shape,X_train.shape,X_test.shape)

(768, 8) (614, 8) (154, 8)


Total data we have is 768 rows. Out of those ,we have taken 80% data for train i.e.614 and 20% of data fro testing i.e. 154.

As a result we have splitted original data into train data and split data